compiler/optimizer/Inliner.cpp

```diff
@@ -1790,7 +1790,11 @@ TR_InlinerBase::addGuardForVirtual(
       {
       createdHCRAndVirtualGuard = true;
 
+      // we merge virtual guards and OSR guards for simplicity in most modes
+      // when using OSR to implement HCR we keep the HCR guards distinct since they
+      // will undergo special processing later in the compilation
       if (virtualGuard &&
+          comp()->getHCRMode() != TR::osr &&
```

Comment:
Okay, so you specifically want the merging to be disabled in this spot for now; in general merging could be allowed after the special processing.